d2/src/main/java/com/linkedin/d2/xds/XdsClient.java

```diff
@@ -0,0 +1,280 @@
+package com.linkedin.d2.xds;
+
+import com.google.protobuf.Value;
+import io.envoyproxy.envoy.config.endpoint.v3.LocalityLbEndpoints;
+import io.envoyproxy.envoy.config.route.v3.VirtualHost;
+import io.envoyproxy.envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public abstract class XdsClient
+{
+  /**
+   * Watcher interface for a single requested xDS resource.
+   */
+  interface ResourceWatcher
+  {
+    // TODO: Handle onError and onResourceNotExist
```

Comment:
Let's add the methods, and we can leave them empty for now with TODOs

Reply:
Added